service/controller/clusterapi/v29/machine_deployment_resource_set.go

```diff
@@ -170,6 +171,20 @@ func NewMachineDeploymentResourceSet(config MachineDeploymentResourceSetConfig)
 		}
 	}
 
+	var tccpSecurityGroupIDResource controller.Resource
+	{
+		c := tccpsecuritygroupid.Config{
+			CMAClient:     config.CMAClient,
+			Logger:        config.Logger,
+			ToClusterFunc: newMachineDeploymentToClusterFunc(config.CMAClient),
```

Comment:
Why do we need the lookup function when the resource is only used within the machine deployment controller?

Reply:
When you check the TCCP outputs resource or template you see that there are no security group IDs. This was about the VPC ID.